CONTRIBUTING.md

```diff
@@ -143,53 +81,110 @@ follow the following steps:
-   guide, before requesting a review. In general, the commits should be squashed
-   into meaningful commits with a helpful message. This means cleanup/fix etc
-   commits should be squashed into the related commit. If you made refactorings
-   it would be best if they are split up into another commit. Think about how
-   a reviewer can best understand your changes. Please follow the
-   [commit message guidelines](#commit-message-guidelines).
-8. After finishing up the squashing, force push your changes to your branch.
+7. If you think you finished the issue, please prepare the branch for review. Please consider our [pull requests and code reviews](https://github.com/camunda/camunda/wiki/Pull-Requests-and-Code-Reviews) guide, before requesting a review. In general, the commits should be squashed into meaningful commits with a helpful message. This means cleanup/fix etc. commits should be squashed into the related commit. If you made refactorings it would be best if they are split up into another commit. Think about how a reviewer can best understand your changes. Please follow the [commit message guidelines](#commit-message-guidelines).
+1. After finishing up the squashing, force push your changes to your branch.
 
    ```
    git push --force-with-lease
    ```
 
+## Build and run Camunda from source
+We are currently working on [architecture streamlining](https://camunda.com/blog/2024/04/simplified-deployment-options-accelerated-getting-started-experience/) to simplify the deployment and build process. While this is in progress, the build instructions are subject to change. The most recent build instructions will always be in this document.
+
+This is a small overview of the contents of this repository:
+- `authentication` - configures authentication for Camunda 8
+- `bom` - bill of materials (BOM) for importing Zeebe dependencies
+- `build-tools` - Zeebe build tools
+- `clients` - client libraries
+- `dist` - provides the Camunda 8 distributions
+- `identity` - component within self-managed Camunda 8 responsible for authentication and authorization
+- `licenses` - the Camunda 8 licenses
+- `monitor` - Monitoring for self-managed Camunda 8
+- `operate` - Monitoring tool for monitoring and troubleshooting processes running in Zeebe
+- `parent` - Parent POM for all Zeebe projects
+- `qa` - quality assurance for Camunda 8
+- `search` - the search clients for Camunda 8 data
+- `service` - internal services for Camunda 8
+- `spring-boot-starter-sdk` - official SDK for Spring Boot
+- `tasklist` - graphical and API application to manage user tasks in Zeebe
+- `testing` - testing libraries for processes and process applications
+- `webapps-common` - shared code between the Camunda 8 web apps
+- `zeebe` - the process automation engine powering Camunda 8
+
+### Build
+
+To **quickly** build all components, run the command: `mvn clean install -DskipTests` in the root folder.
```

Comment:
❓ Why not recommend `mvn clean install -Dquickly` anymore?

The rest of this section implies that building quickly is typically sufficient unless you want a full distribution. However, the current `mvn clean install -DskipTests` provides a full distribution. Note that the `-Dquickly` flag skips all non-essentials goals like assembly of the distribution artifacts. It's useful during development, but not for those that build from source to use the product.

https://github.com/camunda/camunda/blob/f82906ed9e39a6b3305622c26758eb0cf7e314fb/parent/pom.xml#L232-L265

Reply:
I agree. Committing this suggestion. @xomiamoore plesae chime in later if there was another reason for this which we don't see at the moment.